core/src/main/scala/kafka/tools/MirrorMaker.scala

```diff
@@ -701,7 +701,7 @@ object MirrorMaker extends Logging with KafkaMetricsGroup {
   private[tools] object defaultMirrorMakerMessageHandler extends MirrorMakerMessageHandler {
     override def handle(record: BaseConsumerRecord): util.List[ProducerRecord[Array[Byte], Array[Byte]]] = {
       val timestamp: java.lang.Long = if (record.timestamp == Record.NO_TIMESTAMP) null else record.timestamp
-      Collections.singletonList(new ProducerRecord[Array[Byte], Array[Byte]](record.topic, null, timestamp, record.key, record.value))
+      Collections.singletonList(new ProducerRecord[Array[Byte], Array[Byte]](record.topic, record.partition, timestamp, record.key, record.value))
```

Comment:
Why are we changing MM's behavior in this patch? This certainly wouldn't be covered under what the commit message says.

More importantly, this would almost certainly break things for some users as they may be using a different number of partitions in source and destination clusters. The default handler preserves ordering based on keys but allows for a different number of partitions. Normally a different `MirrorMakerMessageHandler` would be swapped in if the user wanted different behavior.